trace-sdk/src/test/java/io/bitrise/trace/data/collector/view/ApplicationForegroundStateDataListenerTest.java

```diff
@@ -0,0 +1,38 @@
+package io.bitrise.trace.data.collector.view;
+
+import android.app.Activity;
+import android.content.Context;
+
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertFalse;
+
+/**
+ * Unit tests for {@link ApplicationForegroundStateDataListener}
+ */
+public class ApplicationForegroundStateDataListenerTest {
```

Comment:
Similarly to [this](ApplicationForegroundStateDataListenerInstrumentedTest), this is already tested implicitly in ApplicationForegroundStateDataListenerInstrumentedTest, except for onStop.

Reply:
Similarly, the coverage report informs they're being missed - https://codecov.io/gh/bitrise-io/trace-android-sdk/src/main/trace-sdk/src/main/java/io/bitrise/trace/data/collector/view/ApplicationForegroundStateDataListener.java - i think these tests are valid.